lib/rubocop/cop/rspec/describe_method.rb

```diff
@@ -26,17 +26,25 @@ class DescribeMethod < Base
         # @!method second_argument(node)
         def_node_matcher :second_argument, <<~PATTERN
           (block
-            (send #rspec? :describe _first_argument $(str _) ...) ...
-          )
+            (send #rspec? :describe _first_argument $_ ...)
+          ...)
+        PATTERN
+
+        # @!method not_method_name?(node)
+        def_node_matcher :not_method_name?, <<~PATTERN
+          {(str !#method_name?) (dstr (str !#method_name?) ...)}
```

Comment:
I usually find it hard to understand boolean logic with negations.
Here it's like "if it's not a string prefixed with # OR not an interpolated string prefixed with #".
And it makes you think - what if it's neither? Like a literal `1`.
And just by looking at the code it's not obvious that we tolerate everything that is not a str/dstr.

I dared pushing a commit making this explicit and removes negation from boolean logic involved. Leaving it up to you if you like it or not.

Reply:
Certainly what you have suggested would be better. Let's go with that. Thank you very much.

Reply:
@pirj I've added you to the CHANGELOG as you have refactored many parts of this PR. Thank you!